pkg/pb/pointerdb.proto

```diff
@@ -50,8 +50,10 @@ message RemoteSegment {
   // TODO: may want to use customtype and fixed-length byte slice
   string piece_id = 2;
   repeated RemotePiece remote_pieces = 3;
-
+  
   bytes merkle_root = 4; // root hash of the hashes of all of these pieces
+  
+  repeated piecestoreroutes.SignedHash remote_pieces_hashes = 5;
```

Comment:
Wondering whether we can avoid storing the signed hashes in the database some how.

Reply:
Yeah, its something I'm concerned about too. I'm not sure but my understanding is that if we calculate merkle tree and save only merkle root in Pointer then we are able to use it to verify single piece hash.

Reply:
I have no idea if we can calculate merkle tree from the piece hashes. But if we keep with the Signed Hashes, they should be just another field of the `RemotePiece`, not a new array under `RemoteSegment`. I.e. Each `RemotePiece` has its `SignedHash` field.

Reply:
> they should be just another field of the `RemotePiece` 
Ok, makes totally make sense

My only concern with keeping all signed hashes is size of DB.

Reply:
I don't think we need to keep the signature. As long as we have verified an incoming hash to be valid, it should be sufficient for storing.

Reply:
but next step will be using hashes in audi. Point 8 from Kaloyan `Prevent uplinks to corrupt storage nodes reputation` doc

Reply:
Makes sense. If validated once, the signature will be valid in future too. No sense to keep it. If not valid, we are not going to store the respective piece in the pointer at all.

Reply:
@mniewrzal Egon's point is that you only need to keep the hash in the Pointer, but not the signature.

Reply:
ah ok :) at some point I had version with just hashes and without storing signature and certs

Reply:
done, only hashes are stored